impala/_thrift_api.py

```diff
@@ -262,7 +298,7 @@ def sendRequestRecvResp(data):
       self.code = self.__http_response.status
       self.message = self.__http_response.reason
       self.headers = self.__http_response.msg
-      self.setAuthCookie()
+      self.setHttpCookies()
```

Comment:
I was wondering if moving L301 to before L318 would make the cookie-handling logic cleaner.

The reason is that in L312 where we check whether we should retry the request, the condition self.areHttpCookiesSaved() was meant to verify if we had any cookies when we made the request the first time. The issue is that by then self.setHttpCookies() in L301 has already updated '__are_matching_cookies_found' and '__http_cookie_dict' with the cookies in the response. I hope I'm making sense :)

What do you think?

Reply:
that makes sense, but I also would prefer to keep the code analyzing the response at one place. How about we return a bool that coveys whether we should retry or not?

Reply:
You are right. Fixed.